tap-agent/src/config.rs

```diff
@@ -343,63 +135,112 @@ impl Cli {
+            [postgres]
+            postgres_port = 5432
+            
+            [network_subgraph]
+            allocation_syncing_interval_ms = 60000
+            recently_closed_allocation_buffer_seconds = 3600
+            
+            [escrow_subgraph]
+            escrow_syncing_interval_ms = 60000
+            
+            [tap]
+            rav_request_trigger_value = 10
+            rav_request_timestamp_buffer_ms = 60000
+            rav_request_timeout_secs = 5
+            rav_request_receipt_limit = 10000
+            max_unnaggregated_fees_per_sender = 20
+        "##;
+
+        // Load the user config file
+        let config_str = std::fs::read_to_string(filename)?;
+
+        // Remove TOML comments, so that we can have shell expansion examples in the file.
+        let config_str = config_str
+            .lines()
+            .filter(|line| !line.trim().starts_with('#'))
+            .collect::<Vec<_>>()
+            .join("\n");
+
+        let config_str = shellexpand::env(&config_str)?;
+
+        let config: Config = Figment::new()
+            .merge(Toml::string(config_defaults))
+            .merge(Toml::string(&config_str))
+            .extract()?;
+
+        Ok(config)
+    }
 }
 
 #[cfg(test)]
 mod tests {
+    use std::fs;
+
+    use serde_assert::{Deserializer, Token};
+
     use super::*;
 
     #[test]
-    fn test_parse_grt_value_to_u64() {
-        assert_eq!(
-            parse_grt_value_to_nonzero_u64("1").unwrap(),
-            1_000_000_000_000_000_000
-        );
-        assert_eq!(
-            parse_grt_value_to_nonzero_u64("1.1").unwrap(),
-            1_100_000_000_000_000_000
-        );
+    fn test_parse_grt_value_to_u128_deserialize() {
+        macro_rules! parse {
+            ($input:expr) => {{
+                let mut deserializer =
+                    Deserializer::builder([Token::Str($input.to_string())]).build();
+                parse_grt_value_to_nonzero_u128(&mut deserializer)
+            }};
+        }
+
+        parse!("1").unwrap();
+
+        assert_eq!(parse!("1").unwrap(), 1_000_000_000_000_000_000);
+        assert_eq!(parse!("1.1").unwrap(), 1_100_000_000_000_000_000);
         assert_eq!(
-            parse_grt_value_to_nonzero_u64("1.000000000000000001").unwrap(),
+            parse!("1.000000000000000001").unwrap(),
             1_000_000_000_000_000_001
         );
+        assert_eq!(parse!("0.000000000000000001").unwrap(), 1);
+        assert!(parse!("0").is_err());
+        assert!(parse!("-1").is_err());
         assert_eq!(
-            parse_grt_value_to_nonzero_u64("0.000000000000000001").unwrap(),
-            1
-        );
-        assert!(parse_grt_value_to_nonzero_u64("0").is_err());
-        assert!(parse_grt_value_to_nonzero_u64("-1").is_err());
-        assert_eq!(
-            parse_grt_value_to_nonzero_u64("1.0000000000000000001").unwrap(),
+            parse!("1.0000000000000000001").unwrap(),
             1_000_000_000_000_000_000
         );
     }
```

Comment:
I like that you are testing here multiple values to check if the deserialization is good. But I'd like to also test the error messages so we are sure that our user will understand and fix the config issue.

Probably something on those lines:
```
assert_eq!(err.to_string(), "GRT value must be greater than 0");
assert_eq!(err.to_string(), "GRT value cannot be represented as a u128 GRT wei value");
```

Reply:
Indeed, good idea!

Reply:
Addressed in [fe728de](https://github.com/graphprotocol/indexer-rs/pull/180/commits/fe728de6b7d5d008006d3bd07b71b6f3bcc0abd3)